ray-operator/controllers/ray/rayjob_controller_test.go

```diff
@@ -197,7 +201,37 @@ var _ = Context("Inside the default namespace", func() {
 			workerPods := corev1.PodList{}
 			Eventually(
 				listResourceFunc(ctx, &workerPods, filterLabels, &client.ListOptions{Namespace: "default"}),
-				time.Second*15, time.Millisecond*500).Should(Equal(3), fmt.Sprintf("workerGroup %v", workerPods.Items))
+				time.Second*15, time.Millisecond*500).Should(Equal(int(*myRayJob.Spec.RayClusterSpec.WorkerGroupSpecs[0].Replicas)), fmt.Sprintf("workerGroup %v", workerPods.Items))
+			if len(workerPods.Items) > 0 {
+				Expect(workerPods.Items[0].Status.Phase).Should(Or(Equal(corev1.PodRunning), Equal(corev1.PodPending)))
+			}
+		})
+
+		It("should increase number of worker pods by updating RayJob", func() {
+			Eventually(
+				getRayClusterNameForRayJob(ctx, myRayJob),
+				time.Second*15, time.Millisecond*500).Should(Not(BeEmpty()), "My RayCluster name  = %v", myRayJob.Status.RayClusterName)
+
+			myRayCluster := &rayiov1alpha1.RayCluster{}
+			Eventually(
+				getResourceFunc(ctx, client.ObjectKey{Name: myRayJob.Status.RayClusterName, Namespace: "default"}, myRayCluster),
+				time.Second*3, time.Millisecond*500).Should(BeNil(), "My myRayCluster  = %v", myRayCluster.Name)
+
+			newReplicas := *myRayCluster.Spec.WorkerGroupSpecs[0].Replicas + 1
+
+			// simulate updating the RayJob directly.
+			err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				*myRayJob.Spec.RayClusterSpec.WorkerGroupSpecs[0].Replicas = newReplicas
+				return k8sClient.Update(ctx, myRayJob)
+			})
+			Expect(err).NotTo(HaveOccurred(), "failed to update RayJob")
+
+			// confirm the number of worker pods is not changed
+			filterLabels := client.MatchingLabels{common.RayClusterLabelKey: myRayJob.Status.RayClusterName, common.RayNodeGroupLabelKey: "small-group"}
+			workerPods := corev1.PodList{}
+			Eventually(
+				listResourceFunc(ctx, &workerPods, filterLabels, &client.ListOptions{Namespace: "default"}),
+				time.Second*15, time.Millisecond*500).Should(Equal(int(newReplicas)), fmt.Sprintf("workerGroup %v", workerPods.Items))
```

Comment:
Why does the number of workers equal newReplicas when the number of worker Pods hasn't changed?

```
// confirm the number of worker pods is not changed
```

The comment seems to be wrong.

Reply:
Thank you for pointing out!